core/alsp_src/generic/bsio.c

```diff
@@ -3812,29 +3906,31 @@ sio_readbuffer()
 #endif /* HAVE_SOCKET */
 
 	case SIO_TYPE_CONSOLE:
-	    nchars = console_io(SIO_FD(buf), (char *)buffer, (size_t)nchars);
+    	    nchars = console_io(SIO_FD(buf), (char *)buffer, (size_t)nchars);
 	    break;
 	   
 	default:
 	    SIO_ERRCODE(buf) = SIOE_INVALIDTYPE;
 	    FAIL;
 	    break;
     }
-
     if (nchars < 0) {
 		if (errno == EINTR)
-	    	SIO_ERRCODE(buf) = SIOE_INTERRUPTED;
+	    		SIO_ERRCODE(buf) = SIOE_INTERRUPTED;
 		else {
-	    	SIO_ERRCODE(buf) = SIOE_SYSCALL;
-	    	SIO_ERRNO(buf) = errno;
+	    		SIO_ERRCODE(buf) = SIOE_SYSCALL;
+	    		SIO_ERRNO(buf) = errno;
 		}
 		FAIL;
     }
     else {
 		SIO_ERRCODE(buf) = SIOE_NORMAL;
 		SIO_LPOS(buf) += nchars;
-		if (nchars == 0)
-	    	SIO_FLAGS(buf) |= SIOF_EOF;
+		if (nchars == 0 && errno != -13)
```

Comment:
Maybe this change is no longer needed, since standard_console_read() no longer sets errno to -13?